lib/spack/spack/solver/asp.py

```diff
@@ -2463,6 +2463,37 @@ def _develop_specs_from_env(spec, env):
     spec.constrain(dev_info["spec"])
 
 
+def _is_reusable_external(packages, spec):
+    pkg_cfg = packages.get(spec.name)
+    if not pkg_cfg:
+        return False
+    externals = pkg_cfg.get("externals")
+    if not externals:
+        return False
+
+    for entry in externals:
+        # Should satisify abstract Spec
+        if not spec.satisfies(spack.spec.Spec(entry["spec"])):
+            continue
+
+        # If a prefix is set, config should have the same.
+        if spec.external_modules:
+            prefix = entry.get("prefix")
+            if (
+                prefix is None
+                or spack.util.path.canonicalize_path(prefix) != spec.external_modules
+            ):
+                continue
```

Comment:
I'm having trouble grokking this block. It looks to me like this is ensuring the prefix matches, but it's keyed by `external_modules` instead of `prefix`. I would've expected something like:

```suggestion
        # If a prefix is set, config should have the same.
        if spec.prefix != entry.get("prefix", ""):
            continue
```

what am I missing here?

Reply:
Not sure what happened there. `prefix` is computed, `external_path` is the right key, maybe that was a weird copy paste issue.